### ISSUE TITLE
updatePod does not update start/end IPs

### ISSUE THREAD

**Issue by resmo** [apache/cloudstack#2797]
I noticed an API break in 4.11 in which updatePod does not update the start and end IPs, introduced by #2048?

Also see #2735

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
4.11
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### STEPS TO REPRODUCE

~~~
 $ cs listPods name=ansible-test-stxtc00039-64183073-pod                     
{
  "count": 1, 
  "pod": [
    {
      "allocationstate": "Enabled", 
      "endip": [
        "10.100.10.254"
      ], 
      "forsystemvms": [
        "0"
      ], 
      "gateway": "10.100.10.1", 
      "id": "b427733b-f0fd-408d-be31-7f2b4e3b2a95", 
      "name": "ansible-test-stxtc00039-64183073-pod", 
      "netmask": "255.255.255.0", 
      "startip": [
        "10.100.10.101"
      ], 
      "vlanid": [
        "vlan://untagged"
      ], 
      "zoneid": "ff7e77a4-103f-4c1c-9878-cea1573cd622", 
      "zonename": "ansible-test-stxtc00039-64183073-zone"
    }
  ]
}
~~~
update the startip

~~~
 $ cs updatePod id=b427733b-f0fd-408d-be31-7f2b4e3b2a95 startip=10.100.10.102
{
  "pod": {
    "allocationstate": "Enabled", 
    "endip": [
      "10.100.10.254"
    ], 
    "forsystemvms": [
      "0"
    ], 
    "gateway": "10.100.10.1", 
    "id": "b427733b-f0fd-408d-be31-7f2b4e3b2a95", 
    "name": "ansible-test-stxtc00039-64183073-pod", 
    "netmask": "255.255.255.0", 
    "startip": [
      "10.100.10.101"
    ], 
    "vlanid": [
      "vlan://untagged"
    ], 
    "zoneid": "ff7e77a4-103f-4c1c-9878-cea1573cd622", 
    "zonename": "ansible-test-stxtc00039-64183073-zone"
  }
}
~~~

##### EXPECTED RESULTS

~~~
startip gets updated to 10.100.10.102
~~~

##### ACTUAL RESULTS

~~~
startip is not updated
~~~

**Comment by kioie** [comment]
Is there a situation where a pod can have multiple IPs as shown below?
```
// Check if start IP and end IP of all the ranges lie in the CIDR subnet.
        final String[] existingPodIpRanges = pod.getDescription().split(",");

        for(String podIpRange: existingPodIpRanges) {
            final String[] existingPodIpRange = podIpRange.split("-");

            if (existingPodIpRange.length > 1) {
                if (!NetUtils.isValidIp4(existingPodIpRange[0]) || !NetUtils.isValidIp4(existingPodIpRange[1])) {
                    continue;
                }

                if (!NetUtils.getCidrSubNet(existingPodIpRange[0], cidrSize).equalsIgnoreCase(NetUtils.getCidrSubNet(cidrAddress, cidrSize))) {
                    throw new InvalidParameterValueException("The start address of the some IP range is not in the CIDR subnet.");
                }

                if (!NetUtils.getCidrSubNet(existingPodIpRange[1], cidrSize).equalsIgnoreCase(NetUtils.getCidrSubNet(cidrAddress, cidrSize))) {
                    throw new InvalidParameterValueException("The end address of the some IP range is not in the CIDR subnet.");
                }

                if (NetUtils.ipRangesOverlap(existingPodIpRange[0], existingPodIpRange[1], gateway, gateway)) {
                    throw new InvalidParameterValueException("The gateway shouldn't overlap some start/end ip addresses");
                }
            }
        }
```
Also, I'm curious as to why the pod start and end IP details are stored as a string in the description field? cc @DaanHoogland @rhtyd 

![Screen Shot 2019-04-23 at 9 29 57 AM](https://user-images.githubusercontent.com/20902920/56559489-77302380-65aa-11e9-8a8a-f9c8cb7b5bf2.png)

**Comment by andrijapanicsb** [comment]
Fixes for this issue are being done in PR #3292  - I'll close the issue here and let's continue the discussion in the PR.
